src/providers/ProfileModalProvider.tsx

```diff
@@ -30,22 +36,26 @@ const initialState: State = {
     statusMsg: '',
   },
   deleteMode: false,
+  screen: '',
   modal: null,
 };
 
-type Action =
-  | { type: ActionType.ShowModal; payload: State };
+type Action = { type: ActionType.ShowModal; payload: State };
 
 interface Props {
   children?: React.ReactElement;
 }
 
 type Reducer = (state: State, action: Action) => State;
 
-const showModal = (dispatch: React.Dispatch<Action>) => (user: Partial<User>, deleteMode: boolean): void => {
+const showModal = (dispatch: React.Dispatch<Action>) => (
+  user: Partial<User>,
+  deleteMode: boolean,
+  screen?: string,
```

Comment:
Try not to give type like this. Use param that could be more commonly used. 

For me, I'd give `onDeleteCallback` type here to customize the callback.

Reply:
@hyochan I'v made some modifications to showModal's params to write the code as you requested. It takes callback methods called [onAddFriend] and [onDeleteFriend] as parameters of showModal, and handles actions after adding or deleting a friend.